apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/service/jpa/h2/ConfigH2InitService.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2023 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.configservice.service.jpa.h2;
+
+import com.ctrip.framework.apollo.biz.service.BizDBPropertySource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.jdbc.datasource.init.DatabasePopulatorUtils;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+
+import javax.annotation.PostConstruct;
+import javax.sql.DataSource;
+
+@Configuration
+@Profile("h2")
+public class ConfigH2InitService {
+
+    @Autowired
+    private BizDBPropertySource dbPropertySource;
+
+    @Autowired
+    private DataSource dataSource;
+
+    @PostConstruct
+    public void runSqlScript() throws Exception {
+        Resource resource = new ClassPathResource("jpa/init.h2.sql");
+        DatabasePopulatorUtils.execute(new ResourceDatabasePopulator(resource), dataSource);
+        dbPropertySource.refresh();
```

Comment:
It doesn't work to manually refresh it as some configs are only loaded once in the startup phase, e.g. [portalSupportedEnvs](https://github.com/apolloconfig/apollo/blob/master/apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/PortalSettings.java#L70).
To make it more compatible, it's better to load the data in an early stage. 
I tried to use the jpa [initialization](https://docs.spring.io/spring-boot/docs/current/reference/html/howto.html#howto.data-initialization) mechanism(data.sql) but got some error saying the table serverconfig not found. There might be some configuration issue but I think this is a direction worth exploring.

```
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Table "ServerConfig" not found (this database is empty); SQL statement:
INSERT INTO "ServerConfig" ("Key", "Cluster", "Value", "Comment", "DataChange_CreatedBy", "DataChange_CreatedTime") VALUES ('eureka.service.url', 'default', 'http://localhost:8080/eureka/', 'Eureka服务Url，多个service以英文逗号分隔', 'default', '1970-01-01 00:00:00'), ('namespace.lock.switch', 'default', 'false', '一次发布只能有一个人修改开关', 'default', '1970-01-01 00:00:00'), ('item.key.length.limit', 'default', '128', 'item key 最大长度限制', 'default', '1970-01-01 00:00:00'), ('item.value.length.limit', 'default', '20000', 'item value最大长度限制', 'default', '1970-01-01 00:00:00'), ('config-service.cache
```

![image](https://user-images.githubusercontent.com/837658/223288038-932cb3f3-8dca-489f-adc7-148b7178ef88.png)

Reply:
@nobodyiam How could we control profile when using data.sql?